src/node/interfaces.cpp

```diff
@@ -168,29 +168,25 @@ class NodeImpl : public Node
     bool getNodesStats(NodesStats& stats) override
     {
         stats.clear();
+        if (!m_context->connman) return false;
 
-        if (m_context->connman) {
-            std::vector<CNodeStats> stats_temp;
-            m_context->connman->GetNodeStats(stats_temp);
-
-            stats.reserve(stats_temp.size());
-            for (auto& node_stats_temp : stats_temp) {
-                stats.emplace_back(std::move(node_stats_temp), false, CNodeStateStats());
-            }
+        std::vector<CNodeStats> stats_temp;
+        m_context->connman->GetNodeStats(stats_temp);
 
+        stats.reserve(stats_temp.size());
+        for (auto& s : stats_temp) {
+            stats.emplace_back(std::move(s), CNodeStateStats());
+        }
+        if (m_context->peerman) {
             // Try to retrieve the CNodeStateStats for each node.
-            if (m_context->peerman) {
-                TRY_LOCK(::cs_main, lockMain);
-                if (lockMain) {
-                    for (auto& node_stats : stats) {
-                        std::get<1>(node_stats) =
-                            m_context->peerman->GetNodeStateStats(std::get<0>(node_stats).nodeid, std::get<2>(node_stats));
-                    }
+            TRY_LOCK(::cs_main, lockMain);
```

Comment:
Is it not the same in `master`?

Reply:
I think in master, we return `fNodeStateStatsAvailable==false` and don't update the fields if we can't get the lock ([code](https://github.com/bitcoin/bitcoin/blob/6863ad79a65842504ab6f5584fac3d1de7ecf87e/src/qt/rpcconsole.cpp#L1200-L1201)) instead of updating them with default-values. `fNodeStateStatsAvailable` is being removed here.